packages/core/scripts/umip-15/2_VoteSimulate.js

```diff
@@ -0,0 +1,215 @@
+    (await voting.getCurrentTime()).toString(),
+    "voting phase",
+    (await voting.getVotePhase()).toNumber(),
+    "CurrentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+  let pendingRequests = await voting.getPendingRequests();
+  assert.equal(pendingRequests.length, 1); // the one proposal should have advanced to a request
+
+  /** *****************************************************
+   * 2) Build vote tx from the foundation wallet         *
+   *******************************************************/
+
+  let currentRoundId = (await voting.getCurrentRoundId()).toString();
+
+  const identifier = pendingRequests[0].identifier.toString();
+  const time = pendingRequests[0].time.toString();
+  const price = web3.utils.toWei("1"); // a "yes" vote
+
+  const salt = getRandomUnsignedInt();
+  // Main net DVM uses the old commit reveal scheme of hashed concatenation of price and salt
+  let voteHash;
+  const request = pendingRequests[0];
+  voteHash = computeVoteHash({
+    price,
+    salt,
+    account: foundationWallet,
+    time: request.time,
+    roundId: currentRoundId,
+    identifier: request.identifier
+  });
+
+  console.log("2. COMMIT VOTE FROM FOUNDATION WALLET\nVote information:");
+  console.table({
+    price: price,
+    salt: salt.toString(),
+    account: foundationWallet,
+    time: time,
+    roundId: currentRoundId,
+    identifier: identifier,
+    voteHash: voteHash
+  });
+
+  /** *****************************************************
+   * 3) Vote on the proposal and validate the commitment *
+   *******************************************************/
+
+  // send the foundation wallet some eth to submit the Tx
+  await web3.eth.sendTransaction({
+    from: accounts[0],
+    to: foundationWallet,
+    value: web3.utils.toWei("1"),
+    gas: 2000000
+  });
+
+  const VoteTx = await voting.commitVote(identifier, time, voteHash, { from: foundationWallet, gas: 2000000 });
+  console.log("Voting Tx done!", VoteTx.tx);
+
+  /** *****************************************************
+   * 4) Advance to the next phase & reveal the vote      *
+   *******************************************************/
+  console.log("3. REVEALING FOUNDATION VOTE");
+  currentTime = (await voting.getCurrentTime()).toNumber();
+  await advanceBlockAndSetTime(web3, currentTime + secondsPerDay);
+
+  console.log(
+    "⏱  Advancing time to move to next voting round to enable reveal\nNew timestamp:",
+    (await voting.getCurrentTime()).toString(),
+    "voting phase",
+    (await voting.getVotePhase()).toNumber(),
+    "currentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+
+  // Once the Voting module has been upgraded in the DVM the round requires a signature to snapshot the current
+  // balances before vote reveals. This enables the script to be re-run after the UMIP-15 upgrade to validate the
+  // new Voting contract has been deployed.
+  if (argv.shouldSignSnapshot) {
```

Comment:
the logic here enables this script to work with both the old and new versions of `Voting.sol`. Once the upgrade is done and you want to re-run the scripts to verify the behavior of the upgraded DVM you will need to add the `--shouldSignSnapshot=true` flag to the execution process to enable the snapshotting before the reveal tx.

Reply:
Nice